httplib.h

```diff
@@ -788,7 +788,7 @@ class Server {
   Server &set_file_request_handler(Handler handler);
 
   Server &set_error_handler(HandlerWithResponse handler);
-  Server &set_error_handler(Handler handler);
+  Server &set_error_handler(const Handler &handler);
```

Comment:
I am not comfortable with changes like this. I intentionally made the parameter as a value as other set handler methods are also taking value params. It creates inconsistency and causes a confusion...

Reply:
I was hesitant about this change too. I feel all should be const references or all should always be moved. Personally, I prefer const references for their low overhead, but since the other calls are perfect as they are, I don't want to change them. I will revert this one.

Reply:
I made a compile-time conditional check to use C++14 init captures if available. This silences the warning on C++14 by moving the handler value into the lambda.